apps/portals/src/Navbar.tsx

```diff
@@ -14,47 +13,75 @@ import {
   preparePostSSORedirect,
   redirectAfterSSO,
 } from 'synapse-react-client/dist/utils/AppUtils'
+import { useEffect, useState } from 'react'
+import { useApplicationSessionContext } from 'synapse-react-client/dist/utils/apputils/session/ApplicationSessionContext'
+import { useLogInDialogContext } from './LogInDialogContext'
+import { useSynapseContext } from 'synapse-react-client/dist/utils/SynapseContext'
+import { useGetCurrentUserProfile } from 'synapse-react-client/dist/utils/hooks/SynapseAPI'
+import { useHistory } from 'react-router-dom'
 
 type SynapseSettingLink = {
   text: string
   hasBorder?: boolean
   settingSubPath?: string
 }
 
-type State = {
-  showMenu: boolean
-}
+const synapseQuickLinks: SynapseSettingLink[] = [
+  {
+    text: 'Profile',
+  },
+  {
+    text: 'Projects',
+    settingSubPath: 'projects',
+  },
+  {
+    text: 'Teams',
+    settingSubPath: 'teams',
+  },
+  {
+    text: 'Challenges',
+    settingSubPath: 'challenges',
+  },
+]
 
-class Navbar extends React.Component<any, State> {
-  synapseQuickLinks: SynapseSettingLink[] = [
-    {
-      text: 'Profile',
-    },
-    {
-      text: 'Projects',
-      settingSubPath: 'projects',
-    },
-    {
-      text: 'Teams',
-      settingSubPath: 'teams',
-    },
-    {
-      text: 'Challenges',
-      settingSubPath: 'challenges',
-    },
-  ]
+function Navbar() {
+  const { accessToken } = useSynapseContext()
+  const isSignedIn = !!accessToken
+  const history = useHistory()
 
-  private openBtnRef = React.createRef<HTMLDivElement>()
+  const { data: userProfile } = useGetCurrentUserProfile()
 
-  constructor(props: any) {
-    super(props)
-    this.state = {
-      showMenu: false,
-    }
+  const [showMenu, setShowMenu] = useState(false)
+  const openBtnRef = React.useRef<HTMLDivElement>(null)
+
+  const { refreshSession, clearSession, twoFactorAuthSSOErrorResponse } =
+    useApplicationSessionContext()
+
+  const { showLoginDialog, setShowLoginDialog } = useLogInDialogContext()
+
```

Comment:
Use context when possible instead of the props that were received when this element was cloned